pkg/datapath/linux/probes/probes.go

```diff
@@ -484,6 +487,78 @@ func HaveOuterSourceIPSupport() (err error) {
 	return nil
 }
 
+// HaveSKBAdjustRoomL2RoomMACSupport tests whether the kernel supports the `bpf_skb_adjust_room` helper
+// with the `BPF_ADJ_ROOM_MAC` mode. To do so, we create a program that requests the passed in SKB
+// to be expanded by 20 bytes. The helper checks the `mode` argument and will return -ENOSUPP if
+// the mode is unknown. Otherwise it should resize the SKB by 20 bytes and return 0.
+func HaveSKBAdjustRoomL2RoomMACSupport() (err error) {
+	defer func() {
+		if err != nil && !errors.Is(err, ebpf.ErrNotSupported) {
+			log.WithError(err).Fatal("failed to probe for bpf_skb_adjust_room L2 room MAC support")
+		}
+	}()
+
+	if err := rlimit.RemoveMemlock(); err != nil {
```

Comment:
This can likely be removed, same with the one in `HaveOuterSourceIPSupport()` above.